zpay32/hophint.go

```diff
@@ -45,3 +45,13 @@ func (h HopHint) Copy() HopHint {
 		CLTVExpiryDelta:           h.CLTVExpiryDelta,
 	}
 }
+
+// HopFee calculates the fee for a given amount that is forwarded over a hop.
+// The amount has to be denoted in milli satoshi. The returned fee is also
+// denoted in milli satoshi.
+func (h HopHint) HopFee(amountMSat int64) int64 {
+	feePpm := int64(h.FeeProportionalMillionths)
+	feeRateMsat := (amountMSat * feePpm) / 1_000_000
```

Comment:
Add 1 here if the division remainder is nonzero. This estimation will underpay in 999_999 cases and depending on how strict the check is on the other side it may fall short.

Reply:
Nice catch!

Reply:
Yeah @bitromortac is right. The [spec](https://github.com/lightning/bolts/blob/master/07-routing-gossip.md#requirements-9) says that we don't really need to handle this rounding issue as nodes are recommended to accept payments that round down.